backend/src/schema/resolvers/posts.js

```diff
@@ -211,10 +225,75 @@ export default {
       })
       return emoted
     },
+    pinPost: async (_parent, params, context, _resolveInfo) => {
+      let pinnedPostWithNestedAttributes
+      const { driver, user } = context
+      const session = driver.session()
+      const { id: userId } = user
+      let writeTxResultPromise = session.writeTransaction(async transaction => {
+        const deletePreviousRelationsResponse = await transaction.run(
+          `
+          MATCH (:User)-[previousRelations:PINNED]->(post:Post)
+          DELETE previousRelations
+          RETURN post
+        `,
+        )
+        return deletePreviousRelationsResponse.records.map(record => record.get('post').properties)
+      })
+      await writeTxResultPromise
+
+      writeTxResultPromise = session.writeTransaction(async transaction => {
+        const pinPostTransactionResponse = await transaction.run(
+          `
+            MATCH (user:User {id: $userId}) WHERE user.role = 'admin'
+            MATCH (post:Post {id: $params.id})
+            MERGE (user)-[pinned:PINNED {createdAt: toString(datetime())}]->(post)
+            RETURN post, pinned.createdAt as pinnedAt
+         `,
+          { userId, params },
+        )
+        return pinPostTransactionResponse.records.map(record => ({
+          pinnedPost: record.get('post').properties,
+          pinnedAt: record.get('pinnedAt'),
+        }))
+      })
+      try {
+        const [transactionResult] = await writeTxResultPromise
+        const { pinnedPost, pinnedAt } = transactionResult
+        pinnedPostWithNestedAttributes = {
+          ...pinnedPost,
+          pinnedAt,
+        }
+      } finally {
+        session.close()
+      }
+      return pinnedPostWithNestedAttributes
+    },
```

Comment:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-10-18T13:22:39Z" title="Friday, October 18th 2019, 3:22:39 pm +02:00">Oct 18, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -211,10 +225,68 @@ export default {
       })
       return emoted
     },
+    pinPost: async (_parent, params, context, _resolveInfo) => {
+      let pinnedPost
+      const { driver, user } = context
+      const session = driver.session()
+      const { id: userId } = user
+      let writeTxResultPromise = session.writeTransaction(async transaction => {
+        const deletePreviousRelationsResponse = await transaction.run(
+          `
+          MATCH ()-[previousRelations:PINNED]->(post:Post)
+          DELETE previousRelations
+          RETURN post
+        `,
+          { params },
+        )
+        return deletePreviousRelationsResponse.records.map(record => record.get('post').properties)
+      })
+      await writeTxResultPromise
+
+      writeTxResultPromise = session.writeTransaction(async transaction => {
+        const pinPostTransactionResponse = await transaction.run(
+          `
+            MATCH (user:User {id: $userId}) WHERE user.role = 'admin'
+            MATCH (post:Post {id: $params.id})
+            MERGE (user)-[:PINNED {createdAt: toString(datetime())}]->(post)
+            RETURN post
+         `,
+          { userId, params },
+        )
+        return pinPostTransactionResponse.records.map(record => record.get('post').properties)
+      })
+      try {
+        ;[pinnedPost] = await writeTxResultPromise
+      } finally {
+        session.close()
+      }
+      return pinnedPost
+    },
+    unpinPost: async (_parent, params, context, _resolveInfo) => {
+      let unpinnedPost
+      const session = context.driver.session()
+      const writeTxResultPromise = session.writeTransaction(async transaction => {
+        const unpinPostTransactionResponse = await transaction.run(
+          `
+          MATCH ()-[previousRelations:PINNED]->(post:Post {id: $params.id})
```


Why not `User` here?
```suggestion
          MATCH (:User)-[previousRelations:PINNED]->(post:Post {id: $params.id})
```

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-18T16:22:29Z" title="Friday, October 18th 2019, 6:22:29 pm +02:00">Oct 18, 2019</time>_

---


same as :point_up_2: